### ISSUE TITLE
Binding failure with spring.cloud.stream.metrics.key

### ISSUE THREAD

**Issue by jvalkeal** [spring-cloud/spring-cloud-dataflow#2118]
Streams apps(2.x snapshots) in dataflow fail to start when metrics are enabled. You can manually see same outside of dataflow by running app manually

```
export SPRING_APPLICATION_JSON='{"SPRING_CLOUD_APPLICATION_GUID":"12974", "spring.cloud.stream.metrics.key":"foostream1.time200.${spring.cloud.application.guid}"}'

java -jar time-source-rabbit-2.0.0.BUILD-SNAPSHOT.jar --spring.cloud.stream.bindings.applicationMetrics.destination=metrics --spring.cloud.stream.metrics.scheduleInterval=60
```
fails with error
```
***************************
APPLICATION FAILED TO START
***************************

Description:

Failed to bind properties under 'spring.cloud.stream.metrics.key' to java.lang.String:

    Property: spring.cloud.stream.metrics.key
    Value: foostream1.time200.${spring.cloud.application.guid}
    Origin: System Environment Property "SPRING_APPLICATION_JSON"
    Reason: Could not resolve placeholder 'spring.cloud.application.guid' in value "foostream1.time200.${spring.cloud.application.guid}"
```

boot2 only resolves environment variable format(XXX_XXX_XXX) from real environment variables, not from other property sources.
So `SPRING_CLOUD_APPLICATION_GUID` would resolve from env variables, but not if it's added to SPRING_APPLICATION_JSON itself.

**Comment by sabbyanandan** [comment]
I ran into the same error last week. The NPE is now addressed in SCSt, but we need to add to the docs re: `spring.cloud.stream.metrics.scheduleInterval`. It is not required today.

So, @jvalkeal, we use SAJ liberally across all the SCDF implementations. Specifically, a lot more in CF and K8S. Will we have to review all the configs then? I can see a "Apps and Boot 2.0" section brewing with such findings and instructions to the users.

**Comment by jvalkeal** [comment]
We miss all variables(having env variable format) what we put in json . i.e. `INSTANCE_INDEX` also is missed if that is referenced using `instance.index`.

**Comment by olegz** [comment]
@sabbyanandan the `spring.cloud.stream.metrics.scheduleInterval` is documented both in javadocs and guide (as of latest update to the docs). Do you think we need to add more?

**Comment by sabbyanandan** [comment]
No, I meant in SCDF. We have been pointing to `spring.cloud.stream.bindings.applicationMetrics.destination` and that was the only setting necessary to use SCSt+Collector today. You can read more about it [here](https://docs.spring.io/spring-cloud-dataflow/docs/1.5.0.BUILD-SNAPSHOT/reference/htmlsingle/#configuration-monitoring-deployed-applications). Without `spring.cloud.stream.metrics.scheduleInterval`, it was failing for me, so it needs to be included as a "note" for 2.0 apps.

**Comment by olegz** [comment]
@sabbyanandan ok, got it. Might as well include the fact that it is now `Duration` and is based on Boot's support for it which means that `2000` or `PT2S`, `2 seconds` etc., are acceptable formats for defining values for this property
